### PR TITLE
[chore] add codeowners to metadata

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This is a minute set of changes, adding codeowners to every component and additional folders as designed by the CODEOWNERS file.

Notable changes are renames:
extension/encodingextension/metadata.yml -> extension/encodingextension/metadata.yaml
cmd/mdatagen/metadata.yaml -> cmd/mdatagen/metadata-sample.yaml

**Link to tracking Issue:**
This is step 2 of #23367

**Testing:**
I have used a tool I created to regenerate CODEOWNERS from this file.

### CODE REVIEW

**Comment by atoulme** [comment]
This is blocked by #24227

File: exporter/opencensusexporter/metadata.yaml
Comment (jpkrohling) [inline exporter/opencensusexporter/metadata.yaml:9]: Can you double-check this one? If this should be a group, shouldn't the group be added to all components?

File: extension/encodingextension/metadata.yaml
Comment (jpkrohling) [inline extension/encodingextension/metadata.yaml:7]: Is this correct? Isn't this part of any distributions?

File: extension/observer/ecsobserver/metadata.yaml
Comment (jpkrohling) [inline extension/observer/ecsobserver/metadata.yaml:9]: Should there be an `@`? Perhaps it's missing a comma?

File: internal/coreinternal/metadata.yaml
Comment (jpkrohling) [inline internal/coreinternal/metadata.yaml:3]: This one also looks odd.

File: internal/filter/metadata.yaml
Comment (jpkrohling) [inline internal/filter/metadata.yaml:3]: And this?

File: internal/sharedcomponent/metadata.yaml
Comment (jpkrohling) [inline internal/sharedcomponent/metadata.yaml:5]: This

File: pkg/translator/jaeger/metadata.yaml
Comment (jpkrohling) [inline pkg/translator/jaeger/metadata.yaml:3]: Here too. Perhaps @frzifus could own this?

File: pkg/translator/opencensus/metadata.yaml
Comment (jpkrohling) [inline pkg/translator/opencensus/metadata.yaml:3]: This

File: receiver/opencensusreceiver/metadata.yaml
Comment (jpkrohling) [inline receiver/opencensusreceiver/metadata.yaml:9]: This

File: testbed/metadata.yaml
Comment (jpkrohling) [inline testbed/metadata.yaml:3]: This

File: extension/encodingextension/metadata.yaml
Comment (atoulme) [reply extension/encodingextension/metadata.yaml:7]: yes, as this is still in development.

File: exporter/opencensusexporter/metadata.yaml
Comment (atoulme) [reply exporter/opencensusexporter/metadata.yaml:9]: Yes, see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L71

The core approvers are apparently codeowners of this component.

Please note the group `@open-telemetry/collector-contrib-approvers` will be added to all components in any case.

File: extension/observer/ecsobserver/metadata.yaml
Comment (atoulme) [reply extension/observer/ecsobserver/metadata.yaml:9]: Good catch!

Comment (atoulme) [reply extension/observer/ecsobserver/metadata.yaml]: Fixed.

File: internal/coreinternal/metadata.yaml
Comment (atoulme) [reply internal/coreinternal/metadata.yaml:3]: Same as above: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L111
The CODEOWNERS file says:
```
internal/coreinternal/                                   @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

File: internal/filter/metadata.yaml
Comment (atoulme) [reply internal/filter/metadata.yaml:3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L113
```
internal/filter/                                         @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

File: internal/sharedcomponent/metadata.yaml
Comment (atoulme) [reply internal/sharedcomponent/metadata.yaml:5]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L118C1-L119C1
```
internal/sharedcomponent/                                @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

File: pkg/translator/jaeger/metadata.yaml
Comment (atoulme) [reply pkg/translator/jaeger/metadata.yaml:3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L130C1-L131C1
```
pkg/translator/jaeger/                                   @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

Sure, if @frzifus is open to the idea, please open an issue.

File: pkg/translator/opencensus/metadata.yaml
Comment (atoulme) [reply pkg/translator/opencensus/metadata.yaml:3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L132
```
pkg/translator/opencensus/                               @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

File: receiver/opencensusreceiver/metadata.yaml
Comment (atoulme) [reply receiver/opencensusreceiver/metadata.yaml:9]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L218C10-L218C28
```
receiver/opencensusreceiver/                             @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

File: testbed/metadata.yaml
Comment (atoulme) [reply testbed/metadata.yaml:3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS#L256C1-L257C1
```
testbed/                                                 @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
```

**Comment by atoulme** [comment]
@jpkrohling thanks for the review, I have replied with more information on each comment. Please let me know if that helps, thanks!

**Comment by atoulme** [comment]
I will update to account for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24424.

File: exporter/opencensusexporter/metadata.yaml
Comment (jpkrohling) [reply exporter/opencensusexporter/metadata.yaml:9]: Ohhh, I missed that this was core's approvers, not contrib. Sounds good, then.

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
> I will update to account for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24424.

Feel free to do it on a follow-up PR if you prefer.

**Comment by atoulme** [comment]
Rebasing off main and regenerating. This PR will grow bigger and badder.

**Comment by jpkrohling** [comment]
Sorry for causing the merge conflict. I fixed the conflicts and pushed the changes to this branch. If you need to change it for some reason, please pull from this branch before doing the next changes, @atoulme.

**Comment by atoulme** [comment]
thanks @jpkrohling all good and ready for review.

**Review by dmitryax — APPROVED** [review]

File: pkg/translator/jaeger/metadata.yaml
Comment (frzifus) [reply pkg/translator/jaeger/metadata.yaml:3]: hm.. y, I can take it - But be prepared for some pings in the future.

**Pull request merged**
